### PR TITLE
Onboard new inter-branch merge flow [Configuration]

### PR DESCRIPTION
**Pull request by f-alizada**
### Context
Onboarding new merge flow based on the github actions

### Changes Made
Added configuration file. The flow was copied from: https://github.com/dotnet/versions/blob/main/Maestro/subscriptions.json. Currently it will not be read or change any behaviour of the merge flow but will be used by the new workflow.

### References
Documentation https://github.com/dotnet/arcade/blob/main/Documentation/Maestro/New-Inter-Branch-Merge-Approach.md
One pager: https://github.com/dotnet/dnceng/blob/main/Documentation/OnePagers/github-action-inter-branch-merge.md

### CODE REVIEW

**Comment by f-alizada** [comment]
Hello @carlossanlop. 
Could you please check this PR/ is this something that you would like to take into the dotnet/runtime? 
The maestro inter-branch merge component will be deprecated soon and will not be creating the PR based on the [configuration](https://github.com/dotnet/versions/blob/main/Maestro/subscriptions.json): 

For more informatnion: 
https://github.com/dotnet/dnceng/blob/main/Documentation/OnePagers/ol-maestro-deprecation.md#component-inter-branch-merge-github-action

File: github-merge-flow.jsonc
Comment (carlossanlop) [inline github-merge-flow.jsonc:16]: Shouldn't this be 9.0-rc?

**Review by carlossanlop — APPROVED** [review]
@f-alizada @mmitche approving with one comment to clarify.

File: github-merge-flow.jsonc
Comment (carlossanlop) [reply github-merge-flow.jsonc:16]: Since we don't yet have release/9.0 or release/9.0-rc* branches, I think we're good for now to merge so we don't get blocked on Monday, but I suspect we need these lines updated.

Comment (f-alizada) [reply github-merge-flow.jsonc:16]: The configuration file was prepared based on the https://github.com/dotnet/versions/blob/main/Maestro/subscriptions.json file. 
If there will be a need for the new onboarding branch, please use current (from the PR) file to update the configuration for additional branches in the merge flow chain :)

**Pull request merged**
